services/FeedbackService.ts

```diff
@@ -17,28 +16,41 @@ export default class FeedbackService {
     this.transactions = new TransactionsManager(entityManager);
   }
 
-  public async getFeedback(canSeeAllFeedback = false, user: UserModel): Promise<PublicFeedback[]> {
-    const feedback = await this.transactions.readOnly(async (txn) => {
+  public async getFeedback(canSeeAllFeedback = false, user: UserModel,
+    options: FeedbackSearchOptions): Promise<PublicFeedback[]> {
+    return this.transactions.readOnly(async (txn) => {
       const feedbackRepository = Repositories.feedback(txn);
-      if (canSeeAllFeedback) return feedbackRepository.getAllFeedback();
-      return feedbackRepository.getAllFeedbackForUser(user);
+      if (canSeeAllFeedback) {
+        return (await feedbackRepository.getAllFeedback(options))
+          .map((fb) => fb.getPublicFeedback());
+      }
+
+      const userFeedback = await feedbackRepository.getAllFeedbackForUser(user);
+      return userFeedback.map((fb) => fb.getPublicFeedback());
     });
-    return feedback.map((fb) => fb.getPublicFeedback());
   }
 
   public async submitFeedback(user: UserModel, feedback: Feedback): Promise<PublicFeedback> {
-    const addedFeedback = await this.transactions.readWrite(async (txn) => {
+    return this.transactions.readWrite(async (txn) => {
+      const event = await Repositories.event(txn).findByUuid(feedback.event);
+      if (!event) throw new NotFoundError('Event not found!');
+
+      const feedbackRepository = Repositories.feedback(txn);
+
+      const hasAlreadySubmittedFeedback = await feedbackRepository.hasUserSubmittedFeedback(user, event);
+      if (hasAlreadySubmittedFeedback) throw new UserError('You have already submitted feedback for this event!');
+
```

Comment:
I just wrapped everything in this top level transaction so that if anything inside fails, the entire transaction fails. I know we're really only relying on reads and then one write inside, but I think this is a better pattern to follow in general